SPECIFICATIONS.md

```diff
@@ -787,13 +789,88 @@ corresponding to the last public key, to sign a new block and attenuate the
 token, or a signature of the last block by the last private key, to seal the
 token.
 
+#### Signed payload generation
+
+The data covered by the signature algorithm depends on the `version` field of
+the `SignedBlock` message. If the field is absent, it defaults to version 0.
+
+##### Version 0 (deprecated)
+
+This defines the block signature payload v0.
+
+The authority block signature payload v0 is the concatenation of:
+- `data_0`: the serialized Datalog
+- `pk_1`: the next public key
+- `alg_1`: the little endian representation of the signature algorithm for `pk_1`
+
+To sign the block at index `n+1`, the signed payload format is the concatenation of:
+- `data_n+1`: the serialized Datalog
+- `pk_n+2`: the next public key
+- `alg_n+2`: the little endian representation of the signature algorithm for `pk_n+2`
+
+if `external_sig_n+1` is present, the signed payload format is instead the concatenation of:
+- `data_n+1`: the serialized Datalog
+- `external_sig_n+1`: the optional external signature of the block
+- `pk_n+2`: the next public key
+- `alg_n+2`: the little endian representation of the signature algorithm for `pk_n+2`
+
+This format is deprecated and will be replaced by version 1 in the future.
+
+the signed payload format for external signatures, thereafter referred as "external signature payload v0", is the concatenation of:
+- `data_n+1`: the serialized Datalog
+- `pk_n+1`: the public key for the next block
+- `alg_n+1`: the little endian representation of the signature algorithm for `pk_n+1`
+
+This format is not supported anymore and should be replaced by version 1.
+
+##### Version 1
+
+This defines the block signature payload v1.
+
+The authority block signature payload v1 is the concatenation of:
+- the binary representation of the ASCII string "\0BLOCK\0"
+- the binary representation of the ASCII string "\0VERSION\0"
+- the little endian representation of the version of the signature payload format
+- the binary representation of the ASCII string "\0PAYLOAD\0"
+- `data_0`: the serialized Datalog
+- the binary representation of the ASCII string "\0ALGORITHM\0"
+- `alg_1`: the little endian representation of the signature algorithm for `pk_1`
+- the binary representation of the ASCII string "\0NEXTKEY\0"
+- `pk_1`: the next public key
+
+To sign the block at index `n+1`, the signed payload format is the concatenation of:
+- the binary representation of the ASCII string "\0BLOCK\0"
+- the binary representation of the ASCII string "\0VERSION\0"
+- the little endian representation of the version of the signature payload format
+- the binary representation of the ASCII string "\0PAYLOAD\0"
+- `data_n+1`: the serialized Datalog
+- the binary representation of the ASCII string "\0PREVSIG\0"
```

Comment:
we could put prevsig after algorithm/nextkey to have a bit more consistency between the authority and attenuation blocks, and group signatures. not a big deal, probably just me being nitpicky

Reply:
done in [5d99d3d](https://github.com/biscuit-auth/biscuit/pull/175/commits/5d99d3d6afb2a019a0732a28a4ff97a442252aa5)